Wikipedia/Code/Session.swift

```diff
@@ -309,15 +314,16 @@ import Foundation
         })
     }
     
-    @objc(getJSONDictionaryFromURL:ignoreCache:completionHandler:) public func getJSONDictionary(from url: URL?, ignoreCache: Bool, completionHandler: @escaping ([String: Any]?, HTTPURLResponse?, Error?) -> Swift.Void) {
-        guard let url = url else {
-            completionHandler(nil, nil, NSError.wmf_error(with: .invalidRequestParameters))
-            return
-        }
-        var request = URLRequest(url: url)
-        if ignoreCache {
-            request.cachePolicy = .reloadIgnoringLocalCacheData
+    @objc(getJSONDictionaryFromURL:withQueryParameters:bodyParameters:ignoreCache:completionHandler:)
+    public func getJSONDictionary(from url: URL?, with queryParameters: [String: Any]? = nil, bodyParameters: [String: Any]? = nil, ignoreCache: Bool, completionHandler: @escaping ([String: Any]?, HTTPURLResponse?, Error?) -> Swift.Void) {
+        guard
+            let url = url,
+            let request = requestWithURL(url: url, queryParameters: queryParameters, bodyParameters: bodyParameters, ignoreCache: ignoreCache)
```

Comment:
is it ok if this `getJSONDictionaryFromURL` uses our `request` wrapper instead of `URLRequest`? with the wrapper we're getting params, user agent etc
if not I'll add a separate method